### PR TITLE
Network fix[V2]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
cleanup and some new utilities added

Signed-off-by: Praveen K Pandey praveen@linux.vnet.ibm.com

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@clebergnu @beraldoleal pushed the changes, please have a look

File: avocado/utils/configure_network.py [hunk 1]
```diff
@@ -43,6 +43,36 @@ class For network interface
     def __init__(self, interface):  # pylint: disable=W0231
         self._interface = interface
 
+    def bring_up(self):
+        """
+         Utility  Used to Bring up interface
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: two spaces here. Also, the case is strange.

File: avocado/utils/configure_network.py [hunk 2]
```diff
@@ -43,6 +43,36 @@ class For network interface
...
+        :return :  True In based on success otherwise Raise NWException
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: True is?

File: avocado/utils/configure_network.py [hunk 3]
```diff
@@ -43,6 +43,36 @@ class For network interface
...
+        cmd = "ifup %s" % self._interface
+        try:
+            process.system(cmd, ignore_status=False, sudo=True)
+            return True
+        except process.CmdError as ex:
+            raise NWException("ifup fails: %s" % ex)
+
+    def bring_down(self):
+        """
+        Utility  Used to Bring down interface
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: same here

File: avocado/utils/configure_network.py [hunk 4]
```diff
@@ -43,6 +43,36 @@ class For network interface
...
+        :return :  True In based on success otherwise Raise NWException
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: same here

File: avocado/utils/configure_network.py [hunk 5]
```diff
@@ -43,6 +43,36 @@ class For network interface
...
+        """
+        cmd = "ifdown %s" % self._interface
+        try:
+            process.system(cmd, sudo=True)
+            return True
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
+
+    def _move_interfaceconfile(self, src_conf, dest_conf):
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: This is generic enough to be renamed to `_move_config_file()` right?

File: avocado/utils/configure_network.py [hunk 6]
```diff
@@ -145,26 +175,55 @@ def is_link_up(self):
...
+
+        :param hwaddr: Pass Hardwae address for defined interface
+        """
+        try:
+            process.system('ifconfig %s hw ether %s' %
+                           (self._interface, hwaddr))
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
+
+    def add_hardware_address(self, maddr):
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: some times you are using "hwaddr" and sometimes "hardware_address", we need to stick with one standard.

File: avocado/utils/configure_network.py [hunk 7]
```diff
@@ -43,6 +43,36 @@ class For network interface
     def __init__(self, interface):  # pylint: disable=W0231
         self._interface = interface
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: what about :

```python3

def __init__(self, if_name):
   self.name = if_name
```

So you don't need the `def get_name()` at the end of this class, right? IIUC, the name should not be private.

File: avocado/utils/configure_network.py [hunk 8]
```diff
@@ -145,26 +175,55 @@ def is_link_up(self):
...
+            return True
         except Exception as ex:
             raise NWException("ifdown fails: %s" % ex)
 
-    def _move_interfaceconfile(self, src_conf, dest_conf):
-        if os.path.exists(src_conf):
-            shutil.move(src_conf, dest_conf)
-        else:
-            raise NWException("%s interface not available" % self._interface)
+    def get_name(self):
```
Comment (beraldoleal) [inline avocado/utils/configure_network.py]: This `get_name` that I was talking about.

**Comment by beraldoleal** [comment]
@PraveenPenguin, since that this is changing the API, it is also recommended to add a small note on the `docs/source/releases/lts/next.rst` file.

**Comment by beraldoleal** [comment]
@PraveenPenguin also, please, take a look on my comments on PR #3594.

It is related and I think that we could merge efforts here.

(sorry about this)

**Comment by PraveenPenguin** [comment]
> @PraveenPenguin also, please, take a look on my comments on PR #3594.
> 
> It is related and I think that we could merge efforts here.
> 
> (sorry about this)

@beraldoleal sure I saw that, may when this PR in we can have other refactor as well may we can split refactor in more PR that easy to maintain as easy to  address review comment, I will work on those as well

**Comment by beraldoleal** [comment]
> > @PraveenPenguin also, please, take a look on my comments on PR #3594.
> > It is related and I think that we could merge efforts here.
> > (sorry about this)
> 
> @beraldoleal sure I saw that, may when this PR in we can have other refactor as well may we can split refactor in more PR that easy to maintain as easy to address review comment, I will work on those as well

IMO, this is a simple adjustment and could be done one a single PR (with multiple commits, of course).

Multiple PRs will introduce multiple APIs changes. But ok, let's move on.

After your changes here I will try to organize all the changes in a new PR, and merge only one.

**Comment by beraldoleal** [comment]
> > @PraveenPenguin also, please, take a look on my comments on PR #3594.
> > It is related and I think that we could merge efforts here.
> > (sorry about this)
> 
> @beraldoleal sure I saw that, may when this PR in we can have other refactor as well may we can split refactor in more PR that easy to maintain as easy to address review comment, I will work on those as well

IMO, this is a simple adjustment and could be done one a single PR (with multiple commits, of course).

Multiple PRs will introduce multiple APIs changes. But ok, let's move on.

After your changes here I will try to organize all the changes in a new PR, and merge only one.

**Comment by PraveenPenguin** [comment]
> > > @PraveenPenguin also, please, take a look on my comments on PR #3594.
> > > It is related and I think that we could merge efforts here.
> > > (sorry about this)
> > 
> > 
> > @beraldoleal sure I saw that, may when this PR in we can have other refactor as well may we can split refactor in more PR that easy to maintain as easy to address review comment, I will work on those as well
> 
> IMO, this is a simple adjustment and could be done one a single PR (with multiple commits, of course).
> 
> Multiple PRs will introduce multiple APIs changes. But ok, let's move on.
> 
> After your changes here I will try to organize all the changes in a new PR, and merge only one.

@beraldoleal thanks for helping, no issues, sure let me try if I can adopt here itself as I started working on V3 , i tried to reach you on avocado irc are you in other  IRC channel

**Comment by PraveenPenguin** [comment]
closing in favour of #3627 @beraldoleal please have a look on that PR

**Pull request closed without merging**